@raphaelcoeffic Running cmake as `cmake -DPCB=X7 -DDISABLE_COMPANION=ON -DCMAKE_BUILD_TYPE=Debug -DLUA=ON -DLUA_COMPILER=ON -DPYTHON_EXECUTABLE=/usr/local/bin/python3 -DCMAKE_VERBOSE_MAKEFILE=ON ../ && make firmware` it stops at:

```
[ 33%] Generating lua_exports_x7.inc
cd /tmp/opentx/build/radio/src/lua && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -E -DCPUARM -DLUA -DLUA_INPUTS -DLUA_EXPORT_GENERATION -DSIMU -I/tmp/opentx/radio/src/targets/taranis -I/tmp/opentx/radio/src/thirdparty -I/tmp/opentx/radio/src/targets/common/arm/stm32 -DPCBTARANIS -DPCBX7 -DRADIO_X7 -DEXPORT /tmp/opentx/radio/src/dataconstants.h > lua_exports_x7.txt
In file included from /tmp/opentx/radio/src/dataconstants.h:24:
In file included from /tmp/opentx/radio/src/targets/taranis/board.h:24:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/11.0.0/include/inttypes.h:30:15: fatal error: 'inttypes.h' file not found
#include_next <inttypes.h>
              ^~~~~~~~~~~~
1 error generated.
make[3]: *** [radio/src/lua/lua_exports_x7.inc] Error 1
make[2]: *** [radio/src/lua/CMakeFiles/lua_export_x7.dir/all] Error 2
make[1]: *** [radio/src/CMakeFiles/firmware.dir/rule] Error 2
make: *** [firmware] Error 2
```

Adding `-DCMAKE_OSX_DEPLOYMENT_TARGET=10.10` results in the same error.

What happens here without this patch is:

- The host C compiler is called in preprocessor only mode (not sure how safe is that for generating the lua exports and why you're not calling the target compiler, but I didn't dive into that)
- The host C compiler autodetected by cmake with Xcode and several SDKs installed is `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc`
- When you invoke the compiler using that command, it has no system includes, just some libc/libc++/swift headers. Hence it fails to find system's `inttypes.h`

The solution in this PR works as:

- cmake already detects `CMAKE_OSX_SYSROOT` automatically and points to the most recent SDK. It's empty on non-macOS systems
- Pass the autodetected system root to the compiler when generating the exports file, so it can find the system headers.

Now, since we're already discussing this... could someone clarify why the compiler targeting the host is being used to run the preprocessor for generating the lua exports file?

--- a/radio/src/CMakeLists.txt
+++ b/radio/src/CMakeLists.txt
@@ -146,7 +146,6 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 include_directories(${CMAKE_CURRENT_BINARY_DIR}/bitmaps/${GUI_DIR})
 
-add_subdirectory(lua)
 include(gui/CMakeLists.txt)
 
 if(RTC_BACKUP_RAM)
@@ -551,3 +550,7 @@ if(NOT MSVC)
   PrintTargetReport("firmware")
 
 endif(NOT MSVC)
+# Include the lua subdirectory after CMAKE_C_COMPILER has been
+# set. Otherwise, the custom commands it includes will use the
+# default CMAKE_C_COMPILER instead of the one we set manually
+add_subdirectory(lua)

